index.ts

```diff
@@ -94,6 +95,17 @@ function handleOptions(_options?: Optionalize<walkSync.Options, 'fs'> | (string|
   return options as  walkSync.Options;
 }
 
+function applyGlobOptions(globs: (string|IMinimatch)[] | undefined, options: MinimatchOptions) { 
+  return globs?.map(glob => { 
+    if (typeof glob === 'string') { 
+      return new Minimatch(glob, options);
+    }
+
+    glob.options = {...glob.options, ...options};
```

Comment:
I'm not sure about munging user provided options here. In general, i would expect a custom minimatch instance to bring its own options and not have them changed.

Reply:
hmm interesting. so what would expected behavior be if i provide an array of instances of `minimatch` as `globs` but ALSO pass in `globOptions`?

Reply:
for example: 
```
{ 
	globs: {new Minimatch('somePattern', options: {nocase: true}},
	globOptions: { dot: true }
}
```

Reply:
in reality, the power of `globOptions` is allowing user to pass in strings and not need to worry about creating instances of minimatch, so i feel like this is an edge case for sure... but obviously still one worth resolving :)

Reply:
Ya, i think the capability of `globOptions` is to configure the `globs` in string form. If one opts into minimatch instances, thats down power-user territory. Additionally, munging using input tends to have unfortunate consequences.

Although I understand why you took this approach, I think the risk of munging user options on minimatch is sufficiently high that we should avoid that.

This should likely be a readme callout, when `globOptions` are documented.

Reply:
gotcha, so if `globs` are instances of minimatch, the globOptions will not be applied. updating